monai/data/ultrasound_confidence_map.py

```diff
@@ -228,17 +246,18 @@ def confidence_laplacian(self, padded_index: NDArray, padded_image: NDArray, bet
         s = self.normalize(s)
 
         # Horizontal penalty
-        s[:vertical_end] += gamma
-        # s[vertical_end:diagonal_end] += gamma * np.sqrt(2) # --> In the paper it is sqrt(2)
-        # since the diagonal edges are longer yet does not exist in the original code
+        s[vertical_end:] += gamma
+        # Here there is a difference between the official MATLAB code and the paper
+        # on the edge penalty. We directly implement what the official code does.
 
         # Normalize differences
         s = self.normalize(s)
 
         # Gaussian weighting function
         s = -(
-            (np.exp(-beta * s, dtype="float64")) + 1.0e-6
-        )  # --> This epsilon changes results drastically default: 1.e-6
+            (np.exp(-beta * s, dtype="float64")) + 10e-6
```

Comment:
I guess it's a default value provided by the paper. Could you please point it out here? Thanks.

Reply:
Hi, sadly there is no mention of this value in the paper, since normally it should serve only as an epsilon value, close to zero, yet in the implementation it does not behave like that at all, I mostly relate this to the problem being not very well conditioned. So in the end, to my best knowledge, this value is just a magic number which changes the results drastically.

That being said, this is exactly how it is implemented in the official MATLAB code: https://campar.in.tum.de/Main/AthanasiosKaramalisCode

I first implemented it as 1.e-6 because 99% of the time you expect it to be 0.000001 yet in this case I also strongly believe that the authors wanted 0.000001 and mistakenly made it 0.00001, so it is a value with lots to speculate about, also made me think a lot about itself, I guess we will never know unless we ask the author, and only if he remembers.